server/index.js

```diff
@@ -35,7 +37,34 @@ app.post('/api/posts', (req, res, next) => {
     .catch(err => next(err));
 });
 
-app.use(staticMiddleware);
+app.get('/api/posts/:postId', (req, res, next) => {
+  const postId = Number(req.params.postId);
+  if (!postId) {
+    throw new ClientError(400, 'postId must be a positive integer');
+  }
+  const sql = `
+    select "imageUrl",
+           "summary",
+           "title",
+           "username",
+           "createdAt",
+           "body"
+    from "posts"
+    join "users" using ("userId")
+    where "postId" = $1
+  `;
+
+  const params = [postId];
+
+  db.query(sql, params)
+    .then(result => {
+      if (!result.rows) {
+        throw new ClientError(400, `cannot find post with postId ${postId}`);
+      }
+      res.json(result.rows);
```

Comment:
Seeing as your client is asking for only one entry, you should send them the one and not an array containing one.